File: app/src/main/java/com/gadgetsfolk/kidsgk/activity/BottomActivity.java
package com.gadgetsfolk.kidsgk.activity;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;

import com.gadgetsfolk.kidsgk.R;
import com.gadgetsfolk.kidsgk.fragment.QuizCategoryFragment;
import com.gadgetsfolk.kidsgk.fragment.QuizFragment;
import com.gadgetsfolk.kidsgk.fragment.VideoFragment;
import com.gadgetsfolk.kidsgk.helper.HelperMethods;
import com.google.android.material.behavior.HideBottomViewOnScrollBehavior;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class BottomActivity extends AppCompatActivity {
    @BindView(R.id.toolbar_title)
    TextView tvToolbar;
    private QuizCategoryFragment quizCategoryFragment;
    private VideoFragment videoFragment;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bottom);

        ButterKnife.bind(this);

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setItemIconTintList(null);
        CoordinatorLayout.LayoutParams layoutParams = (CoordinatorLayout.LayoutParams) navigation.getLayoutParams();
        layoutParams.setBehavior(new HideBottomViewOnScrollBehavior());
        navigation.setOnNavigationItemSelectedListener(listener);
        tvToolbar.setText(getString(R.string.quiz));

        quizCategoryFragment = new QuizCategoryFragment();
        videoFragment = new VideoFragment();

        HelperMethods.loadFragment(quizCategoryFragment, this);

    }

    final BottomNavigationView.OnNavigationItemSelectedListener listener = item -> {
        switch (item.getItemId()){
            case R.id.action_quiz:
                tvToolbar.setText(getString(R.string.quiz));
                HelperMethods.showFragment(quizCategoryFragment, this);
                HelperMethods.hideFragment(videoFragment, this);
                return true;
            case R.id.action_videos:
                tvToolbar.setText(getString(R.string.video));
                HelperMethods.showFragment(videoFragment, this);
                HelperMethods.hideFragment(quizCategoryFragment, this);
                return true;
        }
        return false;
    };
}

File: app/src/main/java/com/gadgetsfolk/kidsgk/activity/BottomActivity.kt
package com.gadgetsfolk.kidsgk.activity

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.gadgetsfolk.kidsgk.R
import com.gadgetsfolk.kidsgk.databinding.ActivityBottomBinding
import com.gadgetsfolk.kidsgk.fragment.QuizCategoryFragment
import com.gadgetsfolk.kidsgk.fragment.VideoFragment
import com.gadgetsfolk.kidsgk.helper.HelperMethods.hideFragment
import com.gadgetsfolk.kidsgk.helper.HelperMethods.loadFragment
import com.gadgetsfolk.kidsgk.helper.HelperMethods.showFragment
import com.google.android.material.behavior.HideBottomViewOnScrollBehavior
import com.google.android.material.bottomnavigation.BottomNavigationView

class BottomActivity : AppCompatActivity() {

    private var quizCategoryFragment: QuizCategoryFragment? = null
    private var videoFragment: VideoFragment? = null

    private lateinit var binding: ActivityBottomBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityBottomBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.itemIconTintList = null
        val layoutParams = navigation.layoutParams as CoordinatorLayout.LayoutParams
        layoutParams.behavior = HideBottomViewOnScrollBehavior<View?>()
        navigation.setOnNavigationItemSelectedListener(listener)
        binding.toolbarTitle.text = getString(R.string.quiz)
        quizCategoryFragment = QuizCategoryFragment()
        videoFragment = VideoFragment()
        loadFragment(quizCategoryFragment, this)
    }

    val listener = BottomNavigationView.OnNavigationItemSelectedListener { item: MenuItem ->
        when (item.itemId) {
            R.id.action_quiz -> {
                binding.toolbarTitle.text = getString(R.string.quiz)
                showFragment(quizCategoryFragment!!, this)
                hideFragment(videoFragment!!, this)
                return@OnNavigationItemSelectedListener true
            }
            R.id.action_videos -> {
                binding.toolbarTitle.text = getString(R.string.video)
                showFragment(videoFragment!!, this)
                hideFragment(quizCategoryFragment!!, this)
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }
}